### ISSUE TITLE
Add FCM

### ISSUE THREAD

**Issue by scouech** [meditohq/medito-app#373]
We need to add FCM. @osamaasifoak I've sent you an invite for Firebase. It's a new project. 
I will design the screen for the value of allowing notifications. 
Once you are working on that ticket, ping me :)

**Comment by scouech** [comment]
@osamaasifoak just to confirm, will the notifications allow us to open the app to a specific screen? 
I guess we need deep linking for that. Let me know how I can help :) Cheers

**Comment by osamaasifoak** [comment]
We don't need deep linking to open a specific screen from a notification.

**Comment by nogweii** [comment]
Note: Firebase was [removed back in May '22](https://github.com/meditohq/medito-app/commit/fba487c8bbfca9fec7dbe017e4fd5ddd3f289797). I think this was in support of #195 and F-Droid team's strict requirements on including apps in the main repo.

If y'all still want FCM, might I suggest looking at supporting [Unified Push](https://unifiedpush.org/)? It has a Flutter library, and supports transparent fallback to FCM. This would allow you to ship a variant of the app without Firebase without losing push notification capabilities.

**Comment by scouech** [comment]
@nogweii Thanks for sharing your thoughts! You make a valid point, and I appreciate your feedback. At the moment, we're focusing on implementing the Firebase version to improve user retention. Notifications have proven to be the most effective method within our current budget limitations. We did encounter some challenges with Unified Push when we tried it a few months ago, and unfortunately, we couldn't release on F-Droid due to token handling issues. However, we're committed to exploring alternative options before the end of the year.
